src/views/TravelRequest/index.js

```diff
@@ -13,48 +13,88 @@ import toaster from "../../helpers/toasts";
 import { toast, ToastContainer, Zoom } from "react-toastify";
 import { useHistory } from "react-router-dom";
 import Axios from "axios";
-import { formatTime } from '../../helpers/timeFormatter';
+import { formatTime } from "../../helpers/timeFormatter";
 import { useTranslation } from "react-i18next";
-
+import { RequestContext } from "../../contexts/RequestContext";
 
 const { REACT_APP_BACKEND_URL } = process.env;
 
-const TravelRequest = () => {
+const TravelRequest = (props) => {
   const classes = useStyles();
   const history = useHistory();
-  const [requestData, setRequestData] = useState({});
+  const userData = JSON.parse(localStorage.getItem("userData"));
+  const requesterName = `${userData.firstname} ${userData.lastname}`;
+
+  const {
+    checkin,
+    checkout,
+    hotelName,
+    hotelId,
+    roomType,
+    roomPrice,
+  } = useContext(RequestContext);
+  const [requestData] = useState({
+    idRoom: props.match.params.reqId,
+    dateStart: checkin,
+    dateEnd: checkout,
+    hotelName,
+  });
+
   const { t } = useTranslation();
 
   const submitrequest = () => {
-    toaster(t("Travel request submitted successfully"), "success");
-    
-    setTimeout(() => {
-      history.push("/booking");
-    }, 4000);
+    const token = localStorage.getItem("userToken");
+    const currentLng = localStorage.getItem("i18nextLng");
+    Axios.post(`${REACT_APP_BACKEND_URL}/requests`, requestData, {
+      headers: {
+        authorization: token,
+        "Accept-Language": currentLng,
+      },
+    })
+      .then((res) => {
+        console.log(res);
+        // setRequestData(res.data.displayRequest);
+        toaster(t("Travel request submitted successfully"), "success");
+        setTimeout(() => {
+          history.push("/dashboard");
+        }, 4000);
+      })
+      .catch((err) => {
+        toaster(err.message, "error");
+      });
+
+    // setTimeout(() => {
+    //   history.push("/booking");
+    // }, 4000);
   };
   const handleEdit = () => {
     setTimeout(() => {
-      history.push("/0/rooms")
-    })
-  }
-  const invalidTime = requestData.dateStart === undefined;
-  const { pathname } = window.location;
-  const requestId = pathname.split('/')[2];
+      history.push(`/${hotelId}/rooms`);
+    });
+  };
+  // const invalidTime = requestData.dateStart === undefined;
+
+  // const { pathname } = window.location;
+  // const requestId = pathname.split('/')[2];
+
+  // useEffect(() => {
+  //   const token = localStorage.getItem('userToken')
+  //   const currentLng = localStorage.getItem('i18nextLng');
+  //   Axios.get(`${REACT_APP_BACKEND_URL}/requests/${requestId}`, {
+  //     headers: {
+  //       'authorization': token,
+  //       "Accept-Language": currentLng
+  //     }
+  //   })
+  //   .then((res) => {
+  //     console.log(res);
+  //     setRequestData(res.data.displayRequest)
+  //   })
+  //   .catch((err) => {
+  //     toaster(err.message, 'error')
+  //   })
+  // },[requestId])
```

Comment:
Remember that here you commented the notification integration where a user when click on a certain notification can see its request info.